### PR TITLE
Add console logging che-server exception messages

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add console error logging on `che-server` failure:
Screenshot 2023-09-05 at 10 58 40

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/22423

### Is it tested? How?

1. Deploy che from `quay.io/ivinokur/che-server:next` image (throws an error on token provisioning).
2. Add a PAT from the dashboard page
3. Refresh the page and see an error.
4. Open the browser console and see the original exception message and trace.

### CODE REVIEW

File: packages/dashboard-frontend/src/store/SanityCheck/index.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export const actionCreators: ActionCreators = {
             delay(1000);
           }
         }
-      } catch (e) {
+      } catch (e: any) {
```
Comment (olexii4) [inline packages/dashboard-frontend/src/store/SanityCheck/index.ts:110]:
```suggestion
      } catch (e: unknown) {
```
TypeScript won't let us perform arbitrary operations on values of type unknown.

WDYT?

File: packages/dashboard-frontend/src/store/SanityCheck/index.ts [hunk 2]
```diff
@@ -118,6 +118,12 @@ export const actionCreators: ActionCreators = {
           type: Type.RECEIVED_BACKEND_CHECK_ERROR,
           error: errorMessage,
         });
+        if (e.message) {
+          console.error(e.message);
+        }
+        if (e.response && e.response.data && e.response.data.trace) {
+          console.error(e.response.data.trace.join('\n'));
```
Comment (olexii4) [inline packages/dashboard-frontend/src/store/SanityCheck/index.ts:125]:
```suggestion
          console.error(common.helpers.errors.getMessage(e));
```

@vinokurig Could you move the next code  `e.response.data.trace.join('\n')` into 'getMessage'  function (https://github.com/eclipse-che/che-dashboard/blob/main/packages/common/src/helpers/errors.ts#L63-L65) ?

Comment (vinokurig) [reply packages/dashboard-frontend/src/store/SanityCheck/index.ts:124]: This is a stack trace list and it is provision error specific, so it makes no sense to put it in ti the common error class.

File: packages/dashboard-frontend/src/store/SanityCheck/index.ts (hunk 1 above)
Comment (vinokurig) [reply packages/dashboard-frontend/src/store/SanityCheck/index.ts:111]: This wont work with the `e.response.data.trace.join('\n')`

Comment (akurinnoy) [reply packages/dashboard-frontend/src/store/SanityCheck/index.ts:111]: you need to use the type guard to guarantee the error object type:

```typescript
        if (
          helpers.errors.includesAxiosResponse(e) &&
          e.response.data.trace &&
          Array.isArray(e.response.data.trace)
        ) {
          console.error(e.response.data.trace.join('\n'));
        }
```

Comment (vinokurig) [reply packages/dashboard-frontend/src/store/SanityCheck/index.ts]: thanks @olexii4, done

**Review by akurinnoy — APPROVED** [review]
LGTM

**Review by ibuziuk — APPROVED** [review]

**Comment by artaleks9** [comment]
* Verified on Eclipse Che with `quay.io/ivinokur/che-server:next` and `quay.io/eclipse/che-dashboard:pr-913` - 
worsk properly
* Screenshots:

**Review by artaleks9 — APPROVED** [review]

**Pull request merged**